### PR TITLE
Support multi-release JARs

### LINKED ISSUE #10756: Quarkus classloader doesn't support multi-release jars
As noted in a recent Quarkus dev mail discussion[1], the `JarClassPathElement` used by the Quarkus classloading infrastructure doesn't currently support multi-release jar files.

[1] https://groups.google.com/d/msg/quarkus-dev/nnKFDiTBlR4/uRksUDDpCAAJ

### PR DESCRIPTION
**Pull request by gastaldi**
Fixes #10756

### CODE REVIEW

**Review by dmlloyd — APPROVED** [review]

File: independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/JarClassPathElement.java [hunk 1]
```diff
@@ -72,9 +73,10 @@ public String getPath() {
                         @Override
                         public URL getUrl() {
                             try {
-                                return new URL("jar", null, jarPath.getProtocol() + ":" + jarPath.getPath() + "!/" + name);
+                                return new URL("jar", null,
+                                        jarPath.getProtocol() + ":" + jarPath.getPath() + "!/" + JarEntries.getRealName(res));
```
Comment (gastaldi) [inline independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/JarClassPathElement.java:77]: This change makes the `io.quarkus.webjar.locator.test.WebJarLocatorDevModeTest` test fail in JDK 8. I'm investigating why that happens

**Review by dmlloyd — APPROVED** [review]

**Comment by gastaldi** [comment]
Damn Windoze :cry:. Something is locking the file and not closing it. Debugging again, but now in my Windows VM

**Comment by gastaldi** [comment]
Shrinkwrap is not closing streams internally, so I had to create a JAR manually for the test

**Comment by gastaldi** [comment]
The problem is with `URL.openStream()`. For some reason this locks the JAR file in Windows.

**Comment by jaikiran** [comment]
From what I remember, @aloubyansky had fixed (or added some workaround for this) when he had run into this while working on packaging related changes. I don't remember what that fix/workaround was.

**Comment by aloubyansky** [comment]
Right, avoid URL.openStream() for local paths.

**Review by jaikiran — APPROVED** [review]

**Pull request merged**